src/main/scala/zio/kafka/serde/Deserializer.scala

```diff
@@ -42,6 +43,9 @@ trait Deserializer[-R, +T] {
    */
   def asTry: Deserializer[R, Try[T]] =
     Deserializer(deserialize(_, _, _).fold(e => Failure(e), v => Success(v)))
+
+  def asOption(implicit @nowarn ev: T <:< AnyRef): Deserializer[R, Option[T]] =
+    Deserializer((topic, headers, data) => deserialize(topic, headers, data).map(Option(_)))
```

Comment:
Small safety suggestion:
```suggestion
    Deserializer((topic, headers, data) => ZIO.foreach(Option(data))(deserialize(topic, headers, _))
```

We can't assume that the underlying deserializer is null-safe.

Reply:
Good one! I will change this.